src/Chapter2.hs

```diff
@@ -75,790 +75,787 @@ When working with lists, the most practical module will be "Data.List":
-values of variables you defined before.
--}
-
-{- |
-=🛡= List implementation
-
-Let's talk a bit about list implementation details. Lists in Haskell
-are implemented as __linked lists__ (or cons-lists). And because
-everything is Haskell is immutable, adding elements at the beginning
-of the lists is cheap. Haskell doesn't need to allocate new memory and
-copy the whole list there; it can just create a new list from a new
-element and a pointer to an already existing list. In other words,
-tails of lists are shared.
-
-For these reasons, adding elements to and extracting elements from the
-beginning of a list is much cheaper and faster than working with the
-end of the list.
-
-In some sense, lists are similar to trains. Let's look at illustration
-of a two-element list:
-
-              . . . . . o o o o o
-  _________    _________   ____  o
- |    y   |   |    x   |   |[]\_n][.
-_|________|_o_|________|_o_|__|____)<
-  oo    oo     oo    oo     oo 00-oo\_
-
-     y      :      x     :  []
-
-You can see that adding new elements (railway carriages) to the left
-is easy: you just need to connect them to the last element in the
-chain.
-
-                            . . . . . o o o o o
-   _________    _________    _________   ____  o
-  |    z   |   |    y   |   |    x   |   |[]\_n][.
- _|________|_o_|________|_o_|________|_o_|__|____)<
-   oo    oo     oo    oo     oo    oo     oo 00-oo\_
-
-       z     :      y      :      x     :  []
-
-But imagine how much difficult it would be to add new carriages to the right?
-
-                            . . . . . o o o o o
-   _________    _________    _________   ____  o
-  |    y   |   |    x   |   |    z   |   |[]\_n][.
- _|________|_o_|________|_o_|________|_o_|__|____)<
-   oo    oo     oo    oo     oo    oo     oo 00-oo\_
-
-       y     :      x      :      z     :  []
-
-You can't simply attach a new carriage anymore. You need to detach the
-locomotive, maybe move trains around the railway a bit for the proper
-position, and only then attach everything back again. The same thing
-with adding elements to the end of the list — it is a slow and costly
-process.
-
--}
-
-{- |
-=⚔️= Task 3
-
-Let's write our first function to process lists in Haskell! Your first
-implementation task is to write a function that returns all elements
-of a list between two given positions inclusive (starting from zero).
-
-Remember that each function returns a new list.
-
->>> subList 3 5 [1 .. 10]
-[4,5,6]
->>> subList 3 0 [True, False, False, True, False]
-[]
-
-♫ NOTE: When implementing, think about various corner cases. You
-  should return an empty list when given numbers are negative.
-
-And also don't forget to check the 'Data.List' module. It is full of
-yummy functions.
-
-Don't forget that you can load this module in GHCi to call functions
-from it!
-
-ghci> :l src/Chapter2.hs
--}
+-- |
+-- =🛡= Lists
+--
+-- __List__ is a crucial data type in Haskell and functional programming
```

Comment:
Would it be possible for you to configure your autoformatter not to change the style of comments?
It's easier for us when the diff is only for solutions and your questions, but not for the whole text :slightly_smiling_face:

Reply:
Sure, let me try, i'm not super familiar with it just was the vs code plugin recommended.